### ISSUE TITLE
Trees in Unity 2019

### ISSUE THREAD

**Issue by ds5678** [ds5678/AssetRipper#3]
There is [interest](https://github.com/mafaca/UtinyRipper/issues/896) in adding an exporter for the [Tree Component](https://docs.unity3d.com/Manual/tree-FirstTree.html).

The source code from Visual Studio looks like this:
```cs
using UnityEngine.Bindings;

namespace UnityEngine
{
    //
    // Summary:
    //     Tree Component for the tree creator.
    [NativeHeader("Modules/Terrain/Public/Tree.h")]
    public sealed class Tree : Component
    {
        public Tree();

        //
        // Summary:
        //     Data asociated to the Tree.
        [NativeProperty("TreeData")]
        public ScriptableObject data { get; set; }
        //
        // Summary:
        //     Tells if there is wind data exported from SpeedTree are saved on this component.
        public bool hasSpeedTreeWind { get; }
    }
}
```
Parts of the Tree.prefab looks like this:
```
Tree:
  m_ObjectHideFlags: 0
  m_CorrespondingSourceObject: {fileID: 0}
  m_PrefabInstance: {fileID: 0}
  m_PrefabAsset: {fileID: 0}
  m_GameObject: {fileID: 4317327013210407872}
  m_SpeedTreeWindAsset: {fileID: 0}
  m_TreeData: {fileID: -2861458992197153078}

MonoBehaviour:
  m_ObjectHideFlags: 0
  m_CorrespondingSourceObject: {fileID: 0}
  m_PrefabInstance: {fileID: 0}
  m_PrefabAsset: {fileID: 0}
  m_GameObject: {fileID: 0}
  m_Enabled: 1
  m_EditorHideFlags: 0
  m_Script: {fileID: 12106, guid: 0000000000000000e000000000000000, type: 0}
  m_Name: Tree Data
  m_EditorClassIdentifier: 
  _uniqueID: 5
  materialHash: 31d6cfe0d16ae931b73c59d7e0c089c0-583008256
  root:
```
And the hex of this Tree component in Windows standard alone build looks like this:
```
00 00 00 00 04 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
```
The size is 24 which could only contain one `Ptr` property (instead of two), and is zero. So they couldn't get the `TreeData` property from it. According to Asset Studio, the solo property is `m_SpeedTreeWindAsset`.

Another problem is that the `MonoBehaviour` with name `Tree Data` has failed to export its script and properties.

In summary, there are issues with exporting the `TreeData` property of the `Tree` Component and the `TreeData` it self.

**Comment by ds5678** [comment]
Maybe 7 days is a little short

**Comment by ds5678** [comment]
This enhancement has been abandoned in favor of a more general approach allowing any built-in component to be exported.
